### PR TITLE
Services: Use per-user portal

### PR DESCRIPTION
**Pull request by LucasChollet**
As introduced by #14608, this PR brings per-user portal for (almost) every service.

Two things are needed to bring this :
 - Support for user detection to create and read the socket at the corresponding place.
 - Configuration changes to tell each service that it has been moved.

1. User detection

To ensure a seamless, multi-user agnostic code. I introduced a variable, `%uid`, to indicate that the path is user-dependent.
As an example, `/tmp/user/%uid/portal/notify` will be replaced by `/tmp/user/100/portal/notify` when being logged as `anon`.

The `/tmp/user` directory is created by root when the first user logs in, and can only be accessed by the superuser. Personal (per-uid) directories are also created by root, but their ownership is forwarded to the user.

There are 3 places where portal paths are used:
 - In `IPC_CLIENT_CONNECTION`, where we connect to the socket
 - In `SystemServer` where we create the socket
 - In each process that unveil it

I wanted to introduce those changes without adding too much overhead. So replacement in those three places are automatically done on the callee side.

2. Configuration changes

The second step is done by all "SystemServer: Launch X at session start-up" commits.

### CODE REVIEW

File: Userland/Services/SystemServer/Service.cpp [hunk 1]
```diff
@@ -35,6 +35,8 @@ ErrorOr<void> Service::setup_socket(SocketDescriptor& socket)
 {
     VERIFY(socket.fd == -1);
 
+    [[maybe_unused]] auto result = Core::System::unlink(socket.path);
```
Comment (linusg) [inline Userland/Services/SystemServer/Service.cpp:38]: Why not `TRY`?

Comment (linusg) [reply Userland/Services/SystemServer/Service.cpp:38]: (preferred way to ignore results is `(void)`)

File: Userland/Services/SystemServer/Service.cpp [hunk 2]
```diff
@@ -209,9 +209,9 @@ void Service::spawn(int socket_fd)
             setenv("SOCKET_TAKEOVER", builder.to_string().characters(), true);
         }
 
-        if (m_account.has_value()) {
+        if (m_account.has_value() && m_account.value().uid() != getuid()) {
             auto& account = m_account.value();
-            if (setgid(account.gid()) < 0 || setgroups(account.extra_gids().size(), account.extra_gids().data()) < 0 || setuid(account.uid()) < 0) {
+            if (!account.login()) {
```
Comment (linusg) [inline Userland/Services/SystemServer/Service.cpp:214]: This {c,sh}ould be a separate commit :)

File: Userland/Services/SystemServer/Service.cpp [hunk 3]
```diff
@@ -320,7 +320,7 @@ Service::Service(Core::ConfigFile const& config, StringView name)
 
         // Need i here to iterate along with all other vectors.
         for (unsigned i = 0; i < socket_paths.size(); i++) {
-            String& path = socket_paths.at(i);
+            String const path = Core::Account::parse_path_with_uid(socket_paths.at(i), m_account.has_value() ? m_account.value().uid() : Optional<uid_t> {});
```
Comment (linusg) [inline Userland/Services/SystemServer/Service.cpp:323]: `auto`

**Review by linusg — COMMENTED** [review]
Love this! I think the commit message prefix should be `Base` instead of `SystemServer` for the config/path-only changes.

File: Userland/Services/SystemServer/Service.cpp (hunk 1 above)
Comment (LucasChollet) [reply Userland/Services/SystemServer/Service.cpp:38]: The purpose of this is to remove potential left-over of previous portal. They are not always there, and unlinking a non-existent path is considered as a failure.

File: Userland/Services/SystemServer/Service.cpp (hunk 2 above)
Comment (LucasChollet) [reply Userland/Services/SystemServer/Service.cpp:214]: I'm removing it as I plan to modify this again when #14296 will land.

File: Userland/Services/SystemServer/Service.cpp (hunk 1 above)
Comment (linusg) [reply Userland/Services/SystemServer/Service.cpp]: That should be a comment IMO, not totally obvious :)

Comment (LucasChollet) [reply Userland/Services/SystemServer/Service.cpp]: Done!
